Userland/Libraries/LibX86/Instruction.h

```diff
@@ -619,9 +712,14 @@ class Instruction {
     u8 m_rep_prefix { 0 };
     OperandSize m_operand_size { OperandSize::Size16 };
     AddressSize m_address_size { AddressSize::Size16 };
+    ProcessorMode m_mode { ProcessorMode::Protected };
     bool m_has_lock_prefix : 1 { false };
     bool m_has_operand_size_override_prefix : 1 { false };
     bool m_has_address_size_override_prefix : 1 { false };
+    bool m_has_rex_w : 1 { false };
+    bool m_has_rex_r : 1 { false };
+    bool m_has_rex_x : 1 { false };
+    bool m_has_rex_b : 1 { false };
```

Comment:
Maybe put these next to the modrm and sib bytes, because the REX prefix seems to act similar to these
than just a `m_has_rex_refix` flag would remain, but haven't read that part of the manual fully

Also using an anonymous unions might ease initialization and overriding of this (same for modrm/sib)

E: Looking further up/down it seems like you have also extended the modrm/sib parts to account for these there so a union would not work anymore....

Reply:
I experimented with a union now and I don't really like how it would turn out. I'd prefer keeping these bools.